asyncpraw/models/reddit/widgets.py

```diff
@@ -342,20 +343,26 @@ def refresh(self):
 
         .. code-block:: python
 
-           widgets = reddit.subreddit("redditdev").widgets
-           widgets.progressive_images = True
-           widgets.refresh()
+            subreddit = await reddit.subreddit("redditdev")
+            widgets = subreddit.widgets
+            widgets.progressive_images = True
+            await widgets.refresh()
 
         """
-        self._fetch()
+        await self._fetch()
 
     def __getattr__(self, attr):
         """Return the value of `attr`."""
         if not attr.startswith("_") and not self._fetched:
-            self._fetch()
-            return getattr(self, attr)
+            raise AttributeError(
+                "{!r} object has no attribute {!r}, did you forget to run '.refresh()'?".format(
+                    self.__class__.__name__, attr
+                )
+            )
         raise AttributeError(
-            "{!r} object has no attribute {!r}".format(self.__class__.__name__, attr)
+            "{!r} object has no attribute {!r}, did you forget to run '.refresh()'?".format(
```

Comment:
Hmm. An exception is raised whenever the user tries to `__getattr__`?

Reply:
This is how it is in praw. Note that `getattr()` does not call `__getattr__()` but instead calls `__getattribute__`.